src/cone_detector.cpp

```diff
@@ -133,167 +285,160 @@ bool ConeDetector::hullIsValid(std::vector<cv::Point>& hull) {
 
 
-    cv::Mat imgThresholded;
-    if (show_step_times_) start = clock();
-    cv::inRange(img_HSV, 
-                cv::Scalar(low_hue_range_, low_saturation_range_, low_value_range_), 
-                cv::Scalar(high_hue_range_, high_saturation_range_, high_value_range_), imgThresholded); //Threshold the image
-    if (show_step_times_) duration_inRange = (std::clock() - start) / (double) CLOCKS_PER_SEC;
+    // Using color detection, convert interesting areas in picture to white and uninteresting areas to black.
+    cv::Mat a_img_thresholded;
+    cv::inRange(img_hsv, 
+                cv::Scalar(alow_hue_range_, alow_saturation_range_, alow_value_range_), 
+                cv::Scalar(ahigh_hue_range_, ahigh_saturation_range_, ahigh_value_range_), a_img_thresholded);
+    cv::Mat b_img_thresholded;
+    cv::inRange(img_hsv, 
+                cv::Scalar(blow_hue_range_, blow_saturation_range_, blow_value_range_), 
+                cv::Scalar(bhigh_hue_range_, bhigh_saturation_range_, bhigh_value_range_), b_img_thresholded);
+
+    // Merge the two filters
+    cv::Mat merged_img_thresholded;
+    cv::bitwise_or(a_img_thresholded, b_img_thresholded, merged_img_thresholded);
 
-    if (show_step_times_) start = clock();
     // Remove small objects from the background.
-    cv::erode(imgThresholded, imgThresholded, getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(3, 3)));
-    cv::dilate(imgThresholded, imgThresholded, getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(3, 3)));
-    cv::GaussianBlur(imgThresholded, imgThresholded, cv::Size(5, 5), 0);
-    cv::Canny(imgThresholded, imgThresholded, 50, 150, 3);
-    if (show_step_times_) duration_erode_dilate_ = (std::clock() - start) / (double) CLOCKS_PER_SEC;
+    cv::erode(merged_img_thresholded, a_img_thresholded, getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(3, 3)));
+    cv::dilate(a_img_thresholded, a_img_thresholded, getStructuringElement(cv::MORPH_ELLIPSE, cv::Size(3, 3)));
+    cv::GaussianBlur(a_img_thresholded, a_img_thresholded, cv::Size(5, 5), 0);
+
+    // Now convert white areas to a white outline.
+    cv::Canny(merged_img_thresholded, merged_img_thresholded, 50, 150, 3);
 
+
+    // Find polygons that are closed loops of white lines.
     std::vector<std::vector<cv::Point> > contours;
     std::vector<cv::Vec4i> hierarchy;
     double contourSize;
     cv::Mat tempImage;
 
-    if (show_step_times_) start = clock();
-    imgThresholded.copyTo(tempImage);
-    if (show_step_times_) duration_copyTo = (std::clock() - start) / (double) CLOCKS_PER_SEC;
-
-    if (show_step_times_) start = clock();
+    merged_img_thresholded.copyTo(tempImage);
     cv::findContours(tempImage, contours, hierarchy, CV_RETR_EXTERNAL, CV_CHAIN_APPROX_SIMPLE);
-    if (show_step_times_) duration_findContours = (std::clock() - start) / (double) CLOCKS_PER_SEC;
 
-    std::vector<cv::Point> *best_cone = NULL;
+    std::vector<cv::Point> *some_polynomial = NULL;
```

Comment:
Again, initialization vs. assignment is identical. I'm happy to change it here. Changed the name to polynomial_of_best_cone. Removed "best_cone" as that's part of future work that I didn't do you.